### PR TITLE
Started work on a Config provider for .NET

### PR DESCRIPTION
**Pull request by csharpfritz**

### CODE REVIEW

File: dotnet/TAML.Core/TamlSerializer.cs [hunk 1]
```diff
@@ -560,6 +583,138 @@ private static bool IsCollectionType(Type type, out Type? elementType)
...
+        foreach (var iface in type.GetInterfaces())
+        {
+            if (iface.IsGenericType && iface.GetGenericTypeDefinition() == typeof(IDictionary<,>))
+            {
+                var args = iface.GetGenericArguments();
+                keyType = args[0];
+                valueType = args[1];
+                return true;
+            }
+        }
```
[the remark this answers is not in the document]
Comment (csharpfritz) [reply dotnet/TAML.Core/TamlSerializer.cs]: @copilot open a new pull request to apply changes based on [this feedback](https://github.com/csharpfritz/Taml/pull/8#discussion_r2637203935)

File: dotnet/TAML.Tests/TamlConfigurationTests.cs [hunk 2]
```diff
@@ -0,0 +1,565 @@
...
+	
+	#endregion
+	
+	#region TamlConfigurationProvider - Edge Cases
+	
+	[Fact]
+	public void GivenNonExistentFile_WhenLoadingConfiguration_ThenConfigurationIsEmpty()
+	{
+		// Given
+		var nonExistentFile = Path.Combine(Path.GetTempPath(), Guid.NewGuid().ToString() + ".taml");
```
[the remark this answers is not in the document]
Comment (csharpfritz) [reply dotnet/TAML.Tests/TamlConfigurationTests.cs]: @copilot open a new pull request to apply changes based on [this feedback](https://github.com/csharpfritz/Taml/pull/8#discussion_r2637203937)

File: dotnet/TAML.Configuration/ConfigurationExtensions.cs [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+	/// <returns></returns>
+	public static IConfigurationBuilder AddTamlConfiguration(this ConfigurationBuilder builder)
+	{
+		return builder.Add(new TamlConfigurationSource("appsettings.taml"));
+	}
+
+	public static IConfigurationBuilder AddTamlConfiguration(this ConfigurationBuilder builder, string path)
+	{
+		return builder.Add(new TamlConfigurationSource(path));
+	}
```
[the remark this answers is not in the document]
Comment (csharpfritz) [reply dotnet/TAML.Configuration/ConfigurationExtensions.cs]: @copilot open a new pull request to apply changes based on [this feedback](https://github.com/csharpfritz/Taml/pull/8#discussion_r2637203926)

File: dotnet/TAML.Configuration/TamlConfigurationProvider.cs [hunk 4]
```diff
@@ -0,0 +1,74 @@
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Primitives;
+using TAML.Core;
+
+namespace TAML.Configuration;
+
+internal class TamlConfigurationProvider(string path) : IConfigurationProvider
```
[the remark this answers is not in the document]
Comment (csharpfritz) [reply dotnet/TAML.Configuration/TamlConfigurationProvider.cs:7]: @copilot open a new pull request to apply changes based on [this feedback](https://github.com/csharpfritz/Taml/pull/8#discussion_r2637203928)

File: dotnet/TAML.Tests/TamlConfigurationTests.cs [hunk 5]
```diff
@@ -0,0 +1,565 @@
...
+			Assert.Equal("MyApplication", config["AppName"]);
+			Assert.Equal("1.0.0", config["Version"]);
+			Assert.Equal("true", config["Debug"]);
+			Assert.Equal("100", config["MaxConnections"]);
+		}
+		finally
+		{
+			File.Delete(tempFile);
+		}
+	}
```
[the remark this answers is not in the document]
Comment (csharpfritz) [reply dotnet/TAML.Tests/TamlConfigurationTests.cs]: @copilot open a new pull request to apply changes based on [this feedback](https://github.com/csharpfritz/Taml/pull/8#discussion_r2637203911)

File: dotnet/TAML.Core/TamlSerializer.cs [hunk 6]
```diff
@@ -560,6 +583,138 @@ private static bool IsCollectionType(Type type, out Type? elementType)
...
+                            // Could be a list of primitives or nested objects
+                            // Check if there are multiple items at the same level
+                            var sameIndentCount = 0;
+                            for (int i = nextIndex; i < lines.Count && lines[i].IndentLevel >= nextLine.IndentLevel; i++)
+                            {
+                                if (lines[i].IndentLevel == nextLine.IndentLevel)
+                                    sameIndentCount++;
+                                if (sameIndentCount > 1)
+                                    break;
+                            }
```
[the remark this answers is not in the document]
Comment (csharpfritz) [reply dotnet/TAML.Core/TamlSerializer.cs]: @copilot open a new pull request to apply changes based on [this feedback](https://github.com/csharpfritz/Taml/pull/8#discussion_r2637203906)

I want you to cache the count

**Comment by csharpfritz** [comment]
@copilot there are some unit tests on the configuration feature that are failing.  investigate and update the recommendations in this PR based on the new code that was merged
